python/src/py/objfloat.c

```diff
@@ -201,6 +201,24 @@ mp_obj_t mp_obj_new_float(mp_float_t value) {
     return MP_OBJ_FROM_PTR(o);
 }
 
+mp_obj_t mp_obj_new_float_via_str(const char* value) {
+    mp_obj_float_t *o = m_new(mp_obj_float_t, 1);
```

Comment:
I don't exactly get what you're doing.
Can't you use this code instead:
`mp_parse_num_decimal(value, strlen(value), false, false, NULL)`
?

Reply:
I firstly need to get the length of the string, until the firsT \X0

Reply:
`strlen` would do that, don't it?